File: src/ConfigServer.Server/Hosting/Internal/EndpointModels/ConfigurationPropertyPayload.cs
using System.Collections.Generic;

namespace ConfigServer.Server
{
    internal class ConfigurationPropertyPayload
    {

        public string PropertyName { get; set; }
        public string PropertyType { get; set; }

        /// <summary>
        /// Display name for property
        /// </summary>
        public string PropertyDisplayName { get; set; }

        /// <summary>
        /// Description for property
        /// </summary>
        public string PropertyDescription { get; set; }

        public Dictionary<string, ConfigurationPropertyPayload> ChildProperty { get; set; }

        public ConfigurationPropertyValidationDefinition ValidationDefinition { get; set; }

        public Dictionary<string, string> Options { get; set; }
    }
}

File: src/ConfigServer.Gui/App/interfaces/configurationPropertyPayload.ts
import { IConfigurationPropertyValidationDefinition } from "./configurationPropertyValidationDefinition";
import { IDictionary } from "./dictionary";

export interface IConfigurationPropertyPayload {
    propertyName: string;
    propertyType: string;
    propertyDisplayName: string;
    propertyDescription: string;
    keyPropertyName: string;
    validationDefinition?: IConfigurationPropertyValidationDefinition;
    options?: IDictionary<string>;

    childProperty?: IDictionary<IConfigurationPropertyPayload>;
}
